src/Aspire.Hosting/Dashboard/DashboardServiceData.cs

```diff
@@ -50,13 +48,7 @@ static GenericResourceSnapshot CreateResourceSnapshot(IResource resource, string
                     ExitCode = snapshot.ExitCode,
                     State = snapshot.State?.Text,
                     StateStyle = snapshot.State?.Style,
-                    HealthState = resource.TryGetAnnotationsIncludingAncestorsOfType<HealthCheckAnnotation>(out _) ? snapshot.HealthStatus switch
```

Comment:
As discussed in our call, this is likely the cause of the incorrect state issues we are seeing in the dashboard. Thinking about it a bit more, its possibly also the reason why we were polling at increased intervals as well (not 100% sure on that though).

Just a reminder to fix this before merging.

Reply:
What did you see?

Reply:
When the resource is created, it has no health reports so appears as "Running". Then the reports arrive, and it transitions to "Running (Unhealthy)", then finally back to "Running".

I'm going to ensure that a resource with a health check annotation that has no reports is still considered unhealthy.

There was an additional behaviour we saw where the health check polling was happening more frequently than we expected it to. I'll fix this and we can check whether things return to what we expected to see.

Reply:
Don't have any health reports should not be the same as the health reports have not arrived yet. So treat null different to 0.

Reply:
This has been fixed in the latest push. Resources that expect health checks have a null health state, which the UI shows as unhealthy.

Resources without health checks appear immediately healthy.

I've also added a placeholder message in the resource details area that appears until the first health data arrives.

See the PR description for new video/screenshots.